docs/user-guide/timeouts/timeouts-main.md

```diff
@@ -0,0 +1,400 @@
+
+Sanity rules are such trivial specifications. For documentation on them, see
+{ref}`sanity <built-in-sanity>` and {ref}`deep sanity <built-in-deep-sanity>`. 
+
+(timeout-causes-library-contracts)=
+#### Library contracts
+
+Some systems are based on multiple library contracts which implement the
+business logic. They also forward storage updates to a single external contract
+holding the storage.
+
+In these systems, it can be appropriate to verify each library independently.
+
+If you encounter timeouts when trying to verify the main entry point contract to
+the system, check the impact of the libraries on the verification by summarizing
+all external library (delegate) calls as `NONDET`, using the option
+`-summarizeExtLibraryCallsAsNonDetPreLinking` as follows:
+
+```sh
+certoraRun ... --prover_args '-summarizeExtLibraryCallsAsNonDetPreLinking true'
+```
+
+```{note}
+This option is only applied for `delegatecall`s and _external_ library calls.
+Internal calls are automatically inlined by the Solidity compiler and are 
+subject to summarizations specified in the spec file's `methods` block.
+```
+
+(timeout-prevention)=
+# Timeout prevention
+
+Timeout prevention approaches fall into these categories.
+1. changing tool settings
+2. changing specs
+3. changing source code
+
+Changing tool settings is least invasive and easy to do, thus it is usually
+preferable to the other options. However, there are cases when parts of the
+input code that are very hard to reason about need to be worked around.
+Sometimes a combination of approaches is needed to resolve a timeout.
+
+
+In the following we will discuss some concrete approaches to timeout prevention.
+This collection will be extended over time based on user's experiences and tool
+improvements.
+
+```{note}
+The old documentation has a section on
+{doc}`troubleshooting </docs/confluence/perplexed>` that addresses timeouts, 
+which might complement the information given here.  
+There is also some helpful information in the section on
+{ref}`summarization <old-summary-example>`.
+Some of the information in these references is out of date.
+```
+
+## Prover settings
+
+In this subsection we list some option combinations that have helped preventing
+timeouts in the past. We group the options by timeout causes they are most
+relevant for. 
+
+### Dealing with a high path count
+
+The Certora Prover internally divides each verification condition into smaller
+subproblems and attempts to solve them separately. This technique is called
+control flow splitting. For a more detailed explanation, see
+{ref}`control-flow-splitting`.
+
+We list a few option combinations that can help in various settings. There is a
+tradeoff between spending time in different places: The Prover can either try to
+spend much time at a low splitting level in the hope that no further splitting
+will be needed, or it can split quickly in the hope that the subproblems will be
+much easier to solve. 
```

Comment:
I'm not sure if this paragraph is clear for laypeople.
If you have time, consider adding here your explanation from the webinar. I think the documentation supports [Graphviz](https://graphviz.org/) graphs, or can be made to support it. I might be able to help with that.

Reply:
I have a PR that explains the splitting, with pictures (no vector graphics so far ... we can discuss that there :-) ) -- it's in line right after this one .. I hope when we have that, things should be clearer; a link to that section is right below